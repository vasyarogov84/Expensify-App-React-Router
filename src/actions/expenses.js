import uuid from 'uuid';

/// Add Expense
export let addExpense = ({ amount = 0, createdAt = 0, description = '', id = 0, note = '' } = {}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        amount,
        createdAt,
        description,
        id:  uuid(),
        note
    }
});
/// Remove Expense
export let removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});
///  Edit Expense
export let editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

