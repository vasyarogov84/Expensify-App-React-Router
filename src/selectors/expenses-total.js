import React from 'react';
import selectExpenses from './expenses';
import { connect } from 'react-redux';

export const ExpenseTotal = (props) => {
   
    return (
        <p>You viewing {props.expenses.length} {props.expenses.length === 1 ? 'expense' : 'expenses'} for are total of {Boolean(props.expenses.length) ?
            (props.expenses.map((expense) => {
                return expense.amount
            }).reduce((a,b) => a + b))/100 : 0}$</p>
    )
}



const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseTotal);
