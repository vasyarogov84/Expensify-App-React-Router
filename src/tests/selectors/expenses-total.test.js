import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseTotal } from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';


test('should show 0 expenses if no expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});


test('should add up multiple expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});



test('should add up single expense', () => {
    const wrapper = shallow(<ExpenseTotal expenses={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});

