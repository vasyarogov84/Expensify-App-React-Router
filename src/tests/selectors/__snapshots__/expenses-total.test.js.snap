// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add up multiple expenses 1`] = `
<p>
  You viewing 
  3
   
  expenses
   for are total of 
  1141.95
  $
</p>
`;

exports[`should add up single expense 1`] = `
<p>
  You viewing 
   
  expenses
   for are total of 
  0
  $
</p>
`;

exports[`should show 0 expenses if no expenses 1`] = `
<p>
  You viewing 
  0
   
  expenses
   for are total of 
  0
  $
</p>
`;
